examples/local/launch_cloud_onprem.py

```diff
@@ -0,0 +1,81 @@
+import argparse
+import os
+import subprocess
+import tempfile
+import textwrap
+import uuid
+import yaml
+
+from click import testing as cli_testing
+
+from sky import cli
+from sky import global_user_state
+from sky.backends import onprem_utils
+from sky.utils import common_utils
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--local-cluster-name',
+                    type=str,
+                    required=True,
+                    help='Name of the local cluster.')
+args = parser.parse_args()
+
+# Public and private keys (autogenerated by Sky)
+public_key = '~/.ssh/sky-key.pub'
+private_key = '~/.ssh/sky-key'
+
+local_cluster_name = args.local_cluster_name
+
+# Sky Task YAML to setup the user 'test'
+sky_yaml_config = {
+    'resources': {
+        'cloud': 'aws'
+    },
+    'file_mounts': {
+        '/user-key': public_key
+    },
+    'setup': textwrap.dedent("""\
+              sudo adduser --disabled-password --gecos '' test
+              sudo -H su test -c 'mkdir ~/.ssh'
+              sudo -H su test -c 'chmod 700 ~/.ssh'
+              sudo -H su test -c 'touch ~/.ssh/authorized_keys'
+              sudo -H su test -c 'chmod 600 ~/.ssh/authorized_keys'
+              sudo -H su test -c 'cat /user-key >> ~/.ssh/authorized_keys'
+            """)
+}
+
+cli_runner = cli_testing.CliRunner()
+onprem_name = f'onprem-cluster-{uuid.uuid4().hex[:6]}'
+with tempfile.NamedTemporaryFile(suffix='.yaml', mode='w') as f:
+    yaml.dump(sky_yaml_config, f)
+    file_path = f.name
+    cli_runner.invoke(cli.launch, ['-c', onprem_name, file_path])
```

Comment:
This failed for me :(
```
(base) romilb@romilbx1yoga:/mnt/d/Romil/Berkeley/Research/sky-experiments/examples/local$ python launch_cloud_onprem.py --local-cluster-name myclus
I 08-26 20:04:35 optimizer.py:605] == Optimizer ==
I 08-26 20:04:35 optimizer.py:628] Estimated cost: $0.4 / hour
I 08-26 20:04:35 optimizer.py:628]
I 08-26 20:04:35 optimizer.py:685] Considered resources (1 node):
I 08-26 20:04:35 optimizer.py:713] ------------------------------------------------------------------
I 08-26 20:04:35 optimizer.py:713]  CLOUD   INSTANCE      vCPUs   ACCELERATORS   COST ($)   CHOSEN
I 08-26 20:04:35 optimizer.py:713] ------------------------------------------------------------------
I 08-26 20:04:35 optimizer.py:713]  AWS     m6i.2xlarge   8       -              0.38          ✔
I 08-26 20:04:35 optimizer.py:713] ------------------------------------------------------------------
I 08-26 20:04:35 optimizer.py:713]
I 08-26 20:04:35 cloud_vm_ray_backend.py:2604] Creating a new cluster: "onprem-cluster-7ebe5a" [1x AWS(m6i.2xlarge)].
I 08-26 20:04:35 cloud_vm_ray_backend.py:2604] Tip: to reuse an existing cluster, specify --cluster (-c). Run `sky status` to see existing clusters.
I 08-26 20:04:45 cloud_vm_ray_backend.py:892] To view detailed progress: tail -n100 -f /home/romilb/sky_logs/sky-2022-08-26-20-04-34-484524/provision.log
I 08-26 20:04:47 cloud_vm_ray_backend.py:1096] Launching on AWS us-east-1 (us-east-1a,us-east-1b,us-east-1c,us-east-1d,us-east-1e,us-east-1f)
I 08-26 20:06:15 log_utils.py:45] Head node is up.
I 08-26 20:07:15 cloud_vm_ray_backend.py:984] Successfully provisioned or found existing VM.
I 08-26 20:07:16 cloud_vm_ray_backend.py:2648] Processing file mounts.
I 08-26 20:07:23 cloud_vm_ray_backend.py:2677] To view detailed progress: tail -n100 -f ~/sky_logs/sky-2022-08-26-20-04-34-484524/file_mounts.log
I 08-26 20:07:23 backend_utils.py:996] Syncing (to 1 node): ~/.ssh/sky-key.pub -> ~/.sky/file_mounts/user-key
I 08-26 20:07:24 cloud_vm_ray_backend.py:1910] Running setup on 1 node.
Adding user `test' ...
Adding new group `test' (1001) ...
Adding new user `test' (1001) with group `test' ...
Creating home directory `/home/test' ...
Copying files from `/etc/skel' ...
I 08-26 20:07:26 cloud_vm_ray_backend.py:1913] Setup completed.
I 08-26 20:07:26 cloud_vm_ray_backend.py:2072] Run commands not specified or empty.
I 08-26 20:07:26 cloud_vm_ray_backend.py:2100]
I 08-26 20:07:26 cloud_vm_ray_backend.py:2100] Cluster name: onprem-cluster-7ebe5a
I 08-26 20:07:26 cloud_vm_ray_backend.py:2100] To log into the head VM: ssh onprem-cluster-7ebe5a
I 08-26 20:07:26 cloud_vm_ray_backend.py:2100] To submit a job:         sky exec onprem-cluster-7ebe5a yaml_file
I 08-26 20:07:26 cloud_vm_ray_backend.py:2100] To stop the cluster:     sky stop onprem-cluster-7ebe5a
I 08-26 20:07:26 cloud_vm_ray_backend.py:2100] To teardown the cluster: sky down onprem-cluster-7ebe5a
NAME                   LAUNCHED     RESOURCES            STATUS  AUTOSTOP  COMMAND                       
onprem-cluster-7ebe5a  16 secs ago  1x AWS(m6i.2xlarge)  UP      -         launch_cloud_onprem.py --...

Local clusters:
NAME          USER  HEAD_IP        RESOURCES    COMMAND
onprem2       test  54.186.202.57  [(no GPUs)]  sky launch -c onprem2 min...
azure-onprem  -     -              -            -
onprem-test   -     -              -            -
[1/4] Checking on-premise environment
E 08-26 20:07:39 subprocess_utils.py:69] 
sky.exceptions.CommandError: Command sudo ray --version failed with return code 1.
Ray is not installed on 54.89.85.189.
Traceback (most recent call last):
  File "launch_cloud_onprem.py", line 72, in <module>
    subprocess.run(f'sky admin deploy {file_path}', shell=True, check=True)
  File "/mnt/d/wsl/anaconda3/lib/python3.7/subprocess.py", line 487, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command 'sky admin deploy /tmp/tmp7_oqop5b.yaml' returned non-zero exit status 1.
```

Reply:
Ah this PR should be tested after #1114 and #1116 are merged!